### PR TITLE
docs: Add information for configuring highlighting

### PR DESCRIPTION
**Pull request by tjdevries**
## Test plan

Doc Updates

### CODE REVIEW

**Review by chrismwendt — APPROVED** [review]

File: doc/dev/how-to/add_support_for_a_language.md [hunk 1]
```diff
@@ -37,6 +37,73 @@ To support precise code intelligence, [write an LSIF indexer](../../code_intelli
...
+
+For example:
+
+```json
+{
+  "syntaxHighlighting": {
+   "languages": {
+     "extensions": {
+        "module": "php",
+        "inc": "php"
```
Comment (varungandhi-src) [inline doc/dev/how-to/add_support_for_a_language.md:71]: What does `"inc"` mean?

File: doc/dev/how-to/add_support_for_a_language.md [hunk 2]
```diff
@@ -37,6 +37,73 @@ To support precise code intelligence, [write an LSIF indexer](../../code_intelli
 
 ## Syntax highlighting support
 
+### Customizing Syntax Highlight Language
```
Comment (varungandhi-src) [inline doc/dev/how-to/add_support_for_a_language.md:40]:
```suggestion
### Customizing syntax highlight language
```

(Matching case convention for other headers.)

File: doc/dev/how-to/add_support_for_a_language.md [hunk 3]
```diff
@@ -37,6 +37,73 @@ To support precise code intelligence, [write an LSIF indexer](../../code_intelli
...
+
+The following settings apply only to the site settings. They are global configuration options for your Sourcegraph instance.
+
+If you have a custom language that is derived from an existing language, it is possible to configure Sourcegraph to highlight that language as another.
```
Comment (varungandhi-src) [inline doc/dev/how-to/add_support_for_a_language.md:44]:
```suggestion
If you have a custom language based on another language that can be highlighted by Sourcegraph, it is possible to configure Sourcegraph to highlight the custom language as the supported language.
```

File: doc/dev/how-to/add_support_for_a_language.md [hunk 4]
```diff
@@ -37,6 +37,73 @@ To support precise code intelligence, [write an LSIF indexer](../../code_intelli
...
+     "extensions": {
+        "strato": "scala"
+      },
+      "patterns": []
+    }
+  }
+}
+```
+
+If you have custom file extensions that map to an existing language, it is possible to configure Sourcegraph to highlight those files as an existing language.
```
Comment (varungandhi-src) [inline doc/dev/how-to/add_support_for_a_language.md:61]:
```suggestion
If you have custom file extensions that map to a supported language, it is possible to configure Sourcegraph to highlight those files as the supported language.
```

File: doc/dev/how-to/add_support_for_a_language.md [hunk 5]
```diff
@@ -37,6 +37,73 @@ To support precise code intelligence, [write an LSIF indexer](../../code_intelli
...
+      },
+      "patterns": []
+    }
+  }
+}
+```
+
+NOTE: In both cases, the `.` is dropped from the file extension.
+
+Additionally, for more complex matching, it possible to pass regexes that will be evaluated (in order listed in the configuration) and if a match is found, will override the syntax highlight language for that file.
```
Comment (varungandhi-src) [inline doc/dev/how-to/add_support_for_a_language.md:81]:
```suggestion
Additionally, for more complex matching, it possible to pass regexes that will be evaluated (in order listed in the configuration) and if a match is found, will override the language used for highlighting that file.
```

Comment (varungandhi-src) [inline doc/dev/how-to/add_support_for_a_language.md:81]:
```suggestion
Additionally, for more complex matching, it possible to pass regexes that will be evaluated (in order listed in the configuration) and if a match is found, will override the language used for highlighting that file.
```

File: doc/dev/how-to/add_support_for_a_language.md [hunk 6]
```diff
@@ -37,6 +37,73 @@ To support precise code intelligence, [write an LSIF indexer](../../code_intelli
...
+          "language": "bash",
+          "pattern": ".bashprofile"
+        }
+      ]
+    }
+  }
+}
+```
+
+### Adding New Syntax Highlighting
```
Comment (varungandhi-src) [inline doc/dev/how-to/add_support_for_a_language.md:105]:
```suggestion
### Adding new syntax highlighters
```

**Pull request merged**